identity-core/src/common/timestamp.rs

```diff
@@ -144,6 +147,27 @@ impl TryFrom<&'_ str> for Timestamp {
   }
 }
 
+// This struct is only used to (potentially) avoid an allocation when deserializing a Timestamp. We cannot deserialize
+// via &str because that breaks serde_json::from_value which we use extensively in the Stronghold bindings.
+// This approach is inspired by https://crates.io/crates/serde_str_helpers, but we only use a subset of the functionality offered by that crate.
+#[derive(Deserialize, Serialize)]
+struct ProvisionalTimestamp<'a>(#[serde(borrow)] Cow<'a, str>);
```

Comment:
Nit: only used for deserialising so maybe remove `Serialize` derive.

```suggestion
#[derive(Deserialize)]
struct ProvisionalTimestamp<'a>(#[serde(borrow)] Cow<'a, str>);
```

Reply:
Would making this `repr(transparent)` or `serde(transparent)` be better?

Reply:
P.S. PR should have the `Wasm` label too since the performance improvement applies to all platforms, probably.